### PR TITLE
feat: use IAM security in DataLakeFileManager 

### PR DESCRIPTION
**Pull request by JesperGomes**
## Description

This PR change a couple of things:
- Use service principle credential in DataLakeFileManager
- Introduce a factory for creating a DataLakeFileManager instance
- Adding a 'storage_account_access' folder 

The PR depends on these two:
- https://github.com/Energinet-DataHub/dh3-infrastructure/pull/217
- https://github.com/Energinet-DataHub/dh3-environments/pull/438

## References

* #902

### CODE REVIEW

File: source/databricks/calculation-engine/package/calculator_job.py [hunk 1]
```diff
@@ -123,20 +123,16 @@ def _start(command_line_args: list[str]) -> None:
...
-        EnvironmentVariable.DATA_STORAGE_ACCOUNT_NAME,
-    ]
-    env_variables = get_env_variables_or_throw(required_env_variables)
-
-    time_zone = env_variables[EnvironmentVariable.TIME_ZONE]
-    storage_account_name = env_variables[EnvironmentVariable.DATA_STORAGE_ACCOUNT_NAME]
+    time_zone = get_env_variable_or_throw(EnvironmentVariable.TIME_ZONE)
+    storage_account_name = get_env_variable_or_throw(
+        EnvironmentVariable.DATA_STORAGE_ACCOUNT_NAME
+    )
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/package/calculator_job.py:129]: Do these belong naturally in this function?

File: source/databricks/calculation-engine/package/calculator_job.py [hunk 2]
```diff
@@ -123,20 +123,16 @@ def _start(command_line_args: list[str]) -> None:
...
 
-    if islocked(storage_account_name, args.data_storage_account_key):
+    if islocked():
```
Comment (BjarkeMeier) [inline source/databricks/calculation-engine/package/calculator_job.py:131]: The way we're implementing the usage of env variables in this PR has some drawbacks.
- a lot of code will be dependent on _how_ configuration is passed to the application
- this makes the code more susceptible to changes
- dependencies become more hidden

If you contemplate e.g. the (ASP).NET way of managing configuration then you'll see it very consequently places the responsibility of application configuration in the host and then delegates configuration through injection (e.g. use of `IOptions`)

File: source/databricks/calculation-engine/package/calculator_job.py (hunk 1 above)
Comment (JesperGomes) [reply source/databricks/calculation-engine/package/calculator_job.py]: now moved outside of this function

File: source/databricks/calculation-engine/package/calculator_job.py (hunk 2 above)
Comment (JesperGomes) [reply source/databricks/calculation-engine/package/calculator_job.py]: I have changed it now so that parameters are injected

**Review by johevemi — APPROVED** [review]

**Pull request merged**